app/models/metric.rb

```diff
@@ -26,7 +26,9 @@ class Metric < ApplicationRecord
 
   acts_as_tree
 
-  attr_protected :service_id, :parent_id, :tenant_id, :audit_ids
+  attr_protected :parent_id, :tenant_id, :audit_ids
+  attr_accessible :service_id, :friendly_name, :system_name, :unit, :description
```

Comment:
> To start from an all-closed default and enable attributes as needed, have a look at [attr_accessible](https://apidock.com/rails/ActiveModel/MassAssignmentSecurity/ClassMethods/attr_accessible).

Meaning you either define protected OR accessible attributes but not both.

Reply:
yes, we can keep one  and that's the reason I removed service_id from protected because this attribute is used while doing mass assignment.

here the attributes  for **attr_protected** and **attr_accessible** are different. 

I think you are saying not to use both together on same model?